### PR TITLE
Update build from source documentation

### LINKED ISSUE #3013: Make permissions requirements clear for gunicorn user
## Problem

As seen in #3002 , a user can correctly follow our docs to build from scratch, but end up with an unusable system since there is not step specifying that the `gunicorn` user needs to be able to write to some subdirectories under the root directory of their Mathesar system. In particular:
- Gunicorn needs to write to the `.media` directory to create files when accepting uploaded TSVs etc.
- Gunicorn needs to read from the `static` directory (I think; someone should double-check my theory on that).

## Proposed solution

Add a step or two in the documentation for building from scratch to either `chown` or `chmod` the directories appropriately.

## Additional context

#3002

### PR DESCRIPTION
**Pull request by deka014**
Fixes #3013 

**Technical details**

**Screenshots**

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dmos62** [comment]
I'll review it, since the relevant issue is part of my backend fixes project.

File: docs/docs/installation/build-from-source/index.md [hunk 1]
```diff
@@ -252,6 +252,12 @@ Then press <kbd>Enter</kbd> to customize this guide with your domain name.
     sudo useradd gunicorn -g gunicorn
     ```
 
+1. Add gunicorn user ownership to .media directory
```
Comment (dmos62) [inline docs/docs/installation/build-from-source/index.md:255]:
```suggestion
1. Make the `gunicorn` user the owner of the `.media` directory
```

**Review by dmos62 — DISMISSED** [review]
Thanks for the PR! I'll merge in a moment after I commit a slight change to wording.

**Comment by dmos62** [comment]
On second thought, @silentninja would you mind giving this a second look?

**Review by silentninja — APPROVED** [review]
Thanks for the PR @deka014! It looks good to me

**Pull request merged**